plugin/src/main/scala/ch/epfl/scala/profilers/ProfilingImpl.scala

```diff
@@ -72,16 +70,20 @@ final class ProfilingImpl[G <: Global](
   def groupPerFile[V](
       kvs: Map[Position, V]
   )(empty: V, aggregate: (V, V) => V): Map[SourceFile, V] = {
-    kvs.groupBy(_._1.source).mapValues {
-      case posInfos: Map[Position, V] => posInfos.valuesIterator.fold(empty)(aggregate)
-    }
+    kvs.groupBy(_._1.source).view.map {
+      case (sf, posInfos: Map[Position, V]) => sf -> posInfos.valuesIterator.fold(empty)(aggregate)
+    }.toMap
```

Comment:
```suggestion
    kvs.groupBy(_._1.source).map {
      case (sf, posInfos: Map[Position, V]) => sf -> posInfos.valuesIterator.fold(empty)(aggregate)
    }
```
Taking a `view` to only perform one operation before converting back with `.toMap` is not useful.